### ISSUE TITLE
Question about using which kinds of count in emptyDrops

### ISSUE THREAD

**Issue by DongzeHE** [MarioniLab/DropletUtils#87]
Hello,

This is a discussion instead of an issue. I put it here because I can't find the discussion tab anywhere. :)

Recently people have started to talk about the unspliced(U), spliced(S), and ambiguous(A) count in single-cell and single-nucleus RNA-sequencing. For example, in the [USA mode](https://www.nature.com/articles/s41592-022-01408-3) we introduced in alevin-fry, we export three gene-by-cell count matrices for spliced, unspliced and ambiguous count, respectively. 

My question is that as emptyDrops and emptyDropsCellRanger takes only **one** matrix, which types of counts should we use to filter empty drops for single-cell and single-nucleus RNA-sequencing datasets? 

It seems like for single-nucleus RNA-sequencing datasets, it is reasonable to sum up U, S, and A count matrix to get an overall count matrix and then feed that into emptyDrops. This is because both spliced and unspliced mRNA molecules are expected to be primed and then sequenced in snRNA-seq. 

However, our concern is whether we should use S+A or U+S+A as the input of emptyDrops for single-cell RNA-sequencing datasets. This is because, in scRNA-seq, although unspliced reads and UMIs are not expected, they are still the product of the priming events that happened in droplets, so they are still valuable and contribute to the droplet profile, and can tell us what's the quality of the droplets. 

For example, Say in a single-cell experiment, a singlet contains a very young cell, and almost everything in this young cell is unspliced. If we consider only the S+A count, this droplet might be filtered by emptyDrops because of the low S+A count. Also, say we have a doublet or even multiple, which only consists of some young cells in a single-cell experiment, their S+A count maybe just at a normal level in that dataset because each of them has a low S+A count. They might be regarded as a singlet and pass the filtering. 

In conclusion, we are pretty sure that we should use U+S+A in single-nucleus, but we are not sure whether we should use S+A or U+S+A count matrix as the input of emptyDrops for scRNA-sequencing datasets. Could you please give us some suggestions or advice? Thanks so much!

Best,
Dongze

**Comment by DongzeHE** [comment]
A follow-up experiment: 
- Count matrices: the unspliced, spliced, and ambiguous count matrix from the 10x PBMC10k dataset.
- Seed: `set.seed(14)`
- FDR threshold: 0.01

When using emptyDrops, 
- 11938 cells passed the filtering when using U+S+A
- 11669 cells passed the filtering when using S+A

When using emptyDropsCellRanger, 
- 11871 cells passed the filtering when using U+S+A 
- 11713 cells passed the filtering when using S+A

**Comment by LTLA** [comment]
In theory, including the unspliced counts should give more power to detect cells from empty droplets, because:

- You have more counts, and thus more detection power to reject the null.
- Unspliced transcripts probably reside in the nucleus and should be less likely to leak into the ambient solution. Thus, the presence of more unspliced transcripts would provide some evidence for cell-containing droplets.

So, U+S+A seems pretty reasonable, assuming independence of counts. There are further distributional assumptions related to the Dirichlet overdispersion, but I don't anticipate that those assumptions are more/less stronger with U+S+A vs S+A. Guess you could have a look at the p-value distributions and check that there is a vague semblence of uniformity among the empty droplets.

> Also, say we have a doublet or even multiple, which only consists of some young cells in a single-cell experiment, their S+A count maybe just at a normal level in that dataset because each of them has a low S+A count. They might be regarded as a singlet and pass the filtering.

Note that this particular scenario doesn't have much relevance to `emptyDrops`, we don't filter to remove doublets anyway. A doublet is still a non-empty droplet as far as we're concerned, so it would have passed regardless of U+S+A vs S+A.
